astropy/nddata/tests/test_ccddata.py

```diff
@@ -1021,3 +1021,13 @@ def test_read_returns_image(tmpdir):
     ccd = CCDData.read(filename, unit='adu')
     # Expecting to get (5, 5), the size of the image
     assert ccd.data.shape == (5, 5)
+
+
+# https://github.com/astropy/astropy/issues/9664
+def test_sliced_ccdata_to_hdu():
+    wcs = WCS(naxis=2)
+    wcs.crpix = 10, 10
+    ccd = CCDData(np.ones((10, 10)), wcs=wcs, unit='pixel')
+    trimmed = ccd[2:-2, 2:-2]
+    hdu = trimmed.to_hdu()
+    assert isinstance(hdu, fits.HDUList)
```

Comment:
Could you maybe check the `CRPIX1` and `CRPIX2` from the header? They should be 8 and 8 (if I'm not mistaken).

Reply:
They are coming out as `-2` apparently:
```
>       assert hdul[0].header['CRPIX1'] == 8
E       assert -2.0 == 8
```
which I do not understand.

Reply:
This is the same behavior as for 3.2 so I am going to just write the test against this.

Reply:
Oh sorry, it should be `wcs.wcs.crpix = 10, 10` 😓

Reply:
oh yeah :man_facepalming: fixed.